docs/marketplace/dev/feature-walkthroughs/202108.0/merchant-switcher-feature-walkthrough.md

```diff
@@ -0,0 +1,26 @@
+---
+title: Merchant Switcher feature walkthrough
+last_updated: Sep 29, 2021
+description: This article provides technical details on the Merchant Switcher feature.
+template: merchant-switcher-feature-walkthrough
+---
+
+With the *Merchant Switcher* feature, the customers can select an active merchant so that he can only see products and offers which belong to this merchant.
+Has been created plugins to check that chosen merchant is equal in all items in Cart, Quote, Checkout modules.
+So if merchant is not equal the item replaces with offer or merchant product from owned to chosen merchant.
+If  the system doesn't have suitable items, a customer will see the error or warning about these items.
+
+## Module dependency graph
+![Entity Diagram](https://confluence-connect.gliffy.net/embed/image/8db03d24-88d4-4715-a5e1-afae4f2ff8ca.png?utm_medium=live&utm_source=confluence)
+
```

Comment:
- MerchantSearch  doesn't belong to this feature
- MerchantSwitcher doesn't depend on Checkout
- MerchantSwitcher provides functionality by introducing some plugins for Checkout, Cart, Wishlist so maybe we need to add extension modules to diagram instead for better context understanding?
- please add an optional module

Reply:
updated.